### ISSUE TITLE
node-gyp rebuild fails on OS X

### ISSUE THREAD

**Issue by kylesimmonds** [atom/atom#12120]
When cloning a fresh fork of the atom repo, I run `npm install` and get the log below. I get similar errors when running `script/build`.

I'm running Node v6.3.0, but I've also tried v5, v4, and v0.12.

    npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
    npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
    npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
    npm WARN deprecated minimatch@0.4.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
    npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
    npm WARN deprecated graceful-fs@3.0.8: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
    npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
    npm WARN prefer global typescript@1.4.1 should be installed with -g
    npm WARN prefer global jasmine-node@1.10.2 should be installed with -g

    > oniguruma@5.1.2 install /Users/username/Development/atom/atom/node_modules/oniguruma
    > node-gyp rebuild

      CC(target) Release/obj.target/oniguruma/deps/onig/regcomp.o
    make: -DNODE_GYP_MODULE_NAME=oniguruma: No such file or directory
    make: *** [Release/obj.target/oniguruma/deps/onig/regcomp.o] Error 1
    gyp ERR! build error 
    gyp ERR! stack Error: `make` failed with exit code: 2
    gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
    gyp ERR! stack     at emitTwo (events.js:106:13)
    gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
    gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
    gyp ERR! System Darwin 15.5.0
    gyp ERR! command "/usr/local/Cellar/node/6.3.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
    gyp ERR! cwd /Users/username/Development/atom/atom/node_modules/oniguruma
    gyp ERR! node -v v6.3.0
    gyp ERR! node-gyp -v v3.3.1
    gyp ERR! not ok 
    atom@1.10.0-dev /Users/username/Development/atom/atom

    [listed packages here]

    npm ERR! Darwin 15.5.0
    npm ERR! argv "/usr/local/Cellar/node/6.3.0/bin/node" "/usr/local/bin/npm" "install"
    npm ERR! node v6.3.0
    npm ERR! npm  v3.10.3
    npm ERR! code ELIFECYCLE

    npm ERR! oniguruma@5.1.2 install: `node-gyp rebuild`
    npm ERR! Exit status 1
    npm ERR! 
    npm ERR! Failed at the oniguruma@5.1.2 install script 'node-gyp rebuild'.
    npm ERR! Make sure you have the latest version of node.js and npm installed.
    npm ERR! If you do, this is most likely a problem with the oniguruma package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     node-gyp rebuild
    npm ERR! You can get information on how to open an issue for this project with:
    npm ERR!     npm bugs oniguruma
    npm ERR! Or if that isn't available, you can get their info via:
    npm ERR!     npm owner ls oniguruma
    npm ERR! There is likely additional logging output above.

    npm ERR! Please include the following file with any support request:
    npm ERR!     /Users/username/Development/atom/atom/npm-debug.log

**Comment by thomasjo** [comment]
Why are you running `npm install`? All you need to do to build Atom is run `script/build`.
Please run `git clean -xfd` (or equivalent), followed simply by `script/build`.

**Comment by kylesimmonds** [comment]
@thomasjo like I said, I get a similar error when running `script/build`, including when I run `script/clean` or `git clean -xfd` like you suggested.

The only difference in running `script/build` is the error is about the minidump package now:

    npm ERR! Failed at the minidump@0.9.0 install script 'node-gyp rebuild'.

**Comment by lee-dohm** [comment]
@kylesimmonds You should:

1. Change directory to the root of your Atom enlistment
1. Run `script/clean`
1. `rm -rf node_modules`
1. Run `script/build`

Then let us know if you run into any more problems.

Also, you should never need to use npm when working on the vast majority of Atom. Using it can screw things up and can cause problems that take a while to dig yourself out from.

**Comment by kylesimmonds** [comment]
@lee-dohm thank you for the help, but it's still throwing errors.

Here's the log:

    Node: v6.3.0
    npm: v3.10.3
    Installing build modules...
    libtool: unrecognized option `-static'
    libtool: Try `libtool --help' for more information.
    make: *** [Release/stackwalk_common.a] Error 1
    gyp ERR! build error 
    gyp ERR! stack Error: `make` failed with exit code: 2
    gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
    gyp ERR! stack     at emitTwo (events.js:106:13)
    gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
    gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
    gyp ERR! System Darwin 15.5.0
    gyp ERR! command "/usr/local/Cellar/node/6.3.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
    gyp ERR! cwd /Users/username/Development/atom/atom/build/node_modules/minidump
    gyp ERR! node -v v6.3.0
    gyp ERR! node-gyp -v v3.3.1
    gyp ERR! not ok 
    npm ERR! Darwin 15.5.0
    npm ERR! argv "/usr/local/Cellar/node/6.3.0/bin/node" "/usr/local/bin/npm" "--userconfig=/Users/username/Development/atom/atom/build/.npmrc" "install" "--loglevel" "error"
    npm ERR! node v6.3.0
    npm ERR! npm  v3.10.3
    npm ERR! code ELIFECYCLE

    npm ERR! minidump@0.9.0 install: `node-gyp rebuild`
    npm ERR! Exit status 1
    npm ERR! 
    npm ERR! Failed at the minidump@0.9.0 install script 'node-gyp rebuild'.
    npm ERR! Make sure you have the latest version of node.js and npm installed.
    npm ERR! If you do, this is most likely a problem with the minidump package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     node-gyp rebuild
    npm ERR! You can get information on how to open an issue for this project with:
    npm ERR!     npm bugs minidump
    npm ERR! Or if that isn't available, you can get their info via:
    npm ERR!     npm owner ls minidump
    npm ERR! There is likely additional logging output above.

    npm ERR! Please include the following file with any support request:
    npm ERR!     /Users/username/Development/atom/atom/build/npm-debug.log

FWIW, I installed Node through Homebrew and used [this method](https://github.com/sindresorhus/guides/blob/master/npm-global-without-sudo.md) to use NPM without sudo. Up until now I haven't had issues installing and using modules.

**Comment by kylesimmonds** [comment]
I just ran a test by installing minidump globally outside of atom, to see if it's an issue with Atom or the minidump module. I got the same error log as above.

I'm going to open an issue on the node-minidump repo and close this one, since it's not an Atom issue. If I find the answer, I'll share it here.

**Comment by kylesimmonds** [comment]
After posting the issue in atom/node-minidump#17, I looked into the error log some more and realized the problems were coming from `libtool`, which is installed by Xcode. I have MAMP installed for traditional LAMP sites, which requires precedence in my `$PATH` and uses its own `libtool`.

To fix it I just renamed the original MAMP `libtool` executable and created a symlink to `/usr/bin/libtool`. This is obviously not a bug with Node, but may be helpful if people have system tools like MAMP installed and are experiencing problems running Node.
